lib/presentation/molecules/youtube_player_molecule.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:flutter/material.dart';
+import 'package:infinite_horizons/presentation/atoms/atoms.dart';
+import 'package:infinite_horizons/presentation/molecules/molecules.dart';
+import 'package:youtube_player_flutter/youtube_player_flutter.dart';
+
+// ignore: must_be_immutable
+class YoutubePlayerMolecule extends StatefulWidget {
+  YoutubePlayerMolecule({required this.url, super.key});
+  final String url;
+  bool isFullScreen = false;
+
+  @override
+  State<YoutubePlayerMolecule> createState() => _YoutubePlayerMoleculeState();
+}
+
+class _YoutubePlayerMoleculeState extends State<YoutubePlayerMolecule> {
+  late final YoutubePlayerController controller;
+  void listener() {
+    widget.isFullScreen = controller.value.isFullScreen;
+    setState(() {});
```

Comment:
Empty set state?

Reply:
Empty setStates are being used to rebuild the pages on variable change

Reply:
This specific one has since been removed